util/resolver/authorizer.go

```diff
@@ -269,13 +269,30 @@ func (ah *authHandler) doBearerAuth(ctx context.Context) (token string, err erro
 	scoped := strings.Join(to.Scopes, " ")
 
 	ah.Lock()
-	if r, exist := ah.scopedTokens[scoped]; exist {
+	for {
+		r, exist := ah.scopedTokens[scoped]
+		if !exist {
+			// no entry cached
+			break
+		}
 		ah.Unlock()
 		r.Wait()
-		if r.expires.IsZero() || r.expires.After(time.Now()) {
+		if r.err != nil {
+			select {
+			case <-ctx.Done():
```

Comment:
this needs a `default:`, otherwise just blocks

Reply:
Good point. Fixed